include/picolibrary/crc.h

```diff
@@ -472,59 +508,180 @@ class Augmented_Byte_Lookup_Table_Calculator {
+        m_xor_output{ parameters.xor_output }
+    {
     }
 
     /**
-     * \brief Feed data into the CRC calculation.
+     * \brief Constructor.
      *
-     * \tparam Iterator Message/augment iterator. The iterated over type must be
-     *         convertible to a std::uint8_t.
+     * \param[in] source The source of the move.
+     */
+    constexpr Direct_Byte_Lookup_Table_Calculator( Direct_Byte_Lookup_Table_Calculator && source ) noexcept = default;
+
+    /**
+     * \brief Constructor.
      *
-     * \param[in] remainder The current CRC calculation remainder.
-     * \param[in] begin The beginning of the message/augment to feed into the calculation.
-     * \param[in] end The end of the message/augment to feed into the calculation.
+     * \param[in] original The original to copy.
+     */
+    constexpr Direct_Byte_Lookup_Table_Calculator( Direct_Byte_Lookup_Table_Calculator const & original ) noexcept = default;
+
+    /**
+     * \brief Destructor.
+     */
+    ~Direct_Byte_Lookup_Table_Calculator() noexcept = default;
+
+    /**
+     * \brief Assignment operator.
      *
-     * \return The resulting CRC calculation remainder.
+     * \param[in] expression The expression to be assigned.
+     *
+     * \return The assigned to object.
+     */
+    constexpr auto operator=( Direct_Byte_Lookup_Table_Calculator && expression ) noexcept
+        -> Direct_Byte_Lookup_Table_Calculator & = default;
+
+    /**
+     * \brief Assignment operator.
+     *
+     * \param[in] expression The expression to be assigned.
+     *
+     * \return The assigned to object.
+     */
+    constexpr auto operator=( Direct_Byte_Lookup_Table_Calculator const & expression ) noexcept
+        -> Direct_Byte_Lookup_Table_Calculator & = default;
+
+    /**
+     * \copydoc picolibrary::CRC::Calculator_Concept::calculate()
      */
     template<typename Iterator>
-    auto feed( Register remainder, Iterator begin, Iterator end ) const noexcept
+    auto calculate( Iterator begin, Iterator end ) const noexcept -> Register
     {
+        auto remainder = m_initial_remainder;
+
         for ( ; begin != end; ++begin ) {
             auto const processed_input = ( *m_process_input )( *begin );
 
+            auto const i = static_cast<std::uint8_t>(
+                               remainder >> ( std::numeric_limits<Register>::digits
+                                              - std::numeric_limits<std::uint8_t>::digits ) )
+                           ^ processed_input;
+
+            remainder <<= std::numeric_limits<std::uint8_t>::digits;
+
+            remainder ^= m_lookup_table[ i ];
+        } // for
+
+        return ( *m_process_output )( remainder ) ^ m_xor_output;
+    }
+
+  private:
+    /**
+     * \brief Calculation lookup table.
+     */
+    Byte_Lookup_Table<Register> m_lookup_table{};
+
+    /**
+     * \brief Calculation preprocessed initial remainder.
+     */
+    Register m_initial_remainder{};
```

Comment:
This member variable should be renamed to reflect that it contains the preprocessed initial remainder, not the specified initial remainder.